src/eZ/ContentView/QueryResultsInjector.php

```diff
@@ -74,7 +77,14 @@ public function injectQueryResults(FilterViewParametersEvent $event)
     private function buildResults(FilterViewParametersEvent $event): iterable
     {
         $view = $event->getView();
-        $content = $view->getContent();
+        if ($view instanceof LocationValueView) {
+            $location = $view->getLocation();
+        }
+
+        if ($view instanceof ContentValueView) {
+            $content = $view->getContent();
+        }
+
```

Comment:
Creating variables under conditions is very error prone. Ideally I'd avoid the spaghetti that follows this block by introducing some strategies. The least you could do for now is to make those variables `null` (e.g. by `:?`) and compare them to `null` later on rather than `isset`

Reply:
I understand... Note that our kernel's ContentViewBuilder has very similar code... :)